index.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const async        = require('async');
+const EventEmitter = require('events');
+const crypto       = require('./lib/crypto.js');
+
+module.exports = class SessionEncrypt extends EventEmitter {
+  constructor(key, store) {
+    super();
+    if (key.length != 32) {
+      throw new Error("Invalid key length: must be 32 bytes");
```

Comment:
http://stackoverflow.com/questions/9864662/how-to-get-the-string-length-in-bytes-in-nodejs/9864762#9864762